test/org/zaproxy/zap/extension/pscanrulesAlpha/XPoweredByHeaderInfoLeakScannerUnitTest.java

```diff
@@ -0,0 +1,111 @@
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import net.htmlparser.jericho.Source;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.utils.I18N;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+
+/**
+ * @author Vahid Rafiei (@vahid_r)
+ * Date 2016-07-06
+ */
+
+@RunWith(MockitoJUnitRunner.class)
+public class XPoweredByHeaderInfoLeakScannerUnitTest {
+
+    private XPoweredByHeaderInfoLeakScanner rule;
+    private PassiveScanThread parent;
+    private List<Alert> alertsRaised;
+
+    @BeforeClass
+    public static void beforeClass() {
+        Constant.messages = Mockito.mock(I18N.class);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        alertsRaised = new ArrayList<>();
+        parent = new PassiveScanThread(null, null, new ExtensionAlert()) {
+            @Override
+            public void raiseAlert(int arg0, Alert arg1) {
+                alertsRaised.add(arg1);
+            }
+        };
+        rule = new XPoweredByHeaderInfoLeakScanner();
+        rule.setParent(parent );
+    }
+
+
+    @Test
+    public void shouldHaveAPluginIdOf10037() throws Exception {
+        // Given
+        int pluginId = rule.getPluginId();
+
+        // When/Then
+        assertThat(pluginId, is(10037));
+    }
+
+    @Test
+    public void shouldNotRaiseAlertIfThereIsNoXPoweredBy() throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                        "Server: Apache-Coyote/1.1\r\n" +
+                        "Content-Type: text/html;charset=ISO-8859-1\r\n" +
+                        "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+
+        // When
+        rule.scanHttpResponseReceive(msg, rule.getPluginId(), source);
```

Comment:
Thats a cnp bug from my code ;) rule.getPluginIn() isnt the right thing to supply here (even though its not checked) - I've changed to use -1